lecture_4/Exercise_2/main.c

```diff
@@ -0,0 +1,97 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef struct node 
+{
+    int val;
+    struct node * next;
+} node_t;
+
+
+void listPrint(const node_t * head) 
+{
+    node_t * current = head;
```

Comment:
my compiler notifies about warring here:
warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
to avoid this you should declare current like this:
const node_t * current = head;